src/x86_64/pe.rs

```diff
@@ -0,0 +1,222 @@
+    }
+
+    fn write_xmm_register(&mut self, _register: pe_unwind_info::x86_64::XmmRegister, _value: u128) {
+        // Ignore
+    }
+}
+
+fn convert_pe_register(r: Register) -> Reg {
+    match r {
+        Register::RAX => Reg::RAX,
+        Register::RCX => Reg::RCX,
+        Register::RDX => Reg::RDX,
+        Register::RBX => Reg::RBX,
+        Register::RSP => Reg::RSP,
+        Register::RBP => Reg::RBP,
+        Register::RSI => Reg::RSI,
+        Register::RDI => Reg::RDI,
+        Register::R8 => Reg::R8,
+        Register::R9 => Reg::R9,
+        Register::R10 => Reg::R10,
+        Register::R11 => Reg::R11,
+        Register::R12 => Reg::R12,
+        Register::R13 => Reg::R13,
+        Register::R14 => Reg::R14,
+        Register::R15 => Reg::R15,
+    }
+}
+
+impl From<&'_ FunctionEpilogInstruction> for OffsetOrPop {
+    fn from(value: &'_ FunctionEpilogInstruction) -> Self {
+        match value {
+            FunctionEpilogInstruction::AddSP(offset) => {
+                if let Ok(v) = (offset / 8).try_into() {
+                    OffsetOrPop::OffsetBy8(v)
+                } else {
+                    OffsetOrPop::None
+                }
+            }
+            FunctionEpilogInstruction::Pop(reg) => OffsetOrPop::Pop(convert_pe_register(*reg)),
+            _ => OffsetOrPop::None,
+        }
+    }
+}
+
+impl From<&'_ UnwindOperation> for OffsetOrPop {
+    fn from(value: &'_ UnwindOperation) -> Self {
+        match value {
+            UnwindOperation::UnStackAlloc(offset) => {
+                if let Ok(v) = (offset / 8).try_into() {
+                    OffsetOrPop::OffsetBy8(v)
+                } else {
+                    OffsetOrPop::None
+                }
+            }
+            UnwindOperation::PopNonVolatile(reg) => OffsetOrPop::Pop(convert_pe_register(*reg)),
+            _ => OffsetOrPop::None,
+        }
+    }
+}
+
+impl PeUnwinding for ArchX86_64 {
+    fn unwind_frame<F, D>(
+        sections: PeSections<D>,
+        address: u32,
+        regs: &mut Self::UnwindRegs,
+        read_stack: &mut F,
+    ) -> Result<UnwindResult<Self::UnwindRule>, PeUnwinderError>
+    where
+        F: FnMut(u64) -> Result<u64, ()>,
+        D: std::ops::Deref<Target = [u8]>,
+    {
+        let entries = FunctionTableEntries::parse(sections.pdata);
+        let Some(function) = entries.lookup(address) else {
+            return Ok(UnwindResult::ExecRule(UnwindRuleX86_64::JustReturn));
+        };
+
+        let read_stack_err = |read_stack: &mut F, addr| {
+            read_stack(addr).map_err(|()| PeUnwinderError::MissingStackData(Some(addr)))
+        };
+
+        let unwind_info_address = function.unwind_info_address.get();
+        let unwind_info =
+            UnwindInfo::parse(sections.unwind_info_memory_at_rva(unwind_info_address)?)
+                .ok_or(PeUnwinderError::UnwindInfoParseError)?;
+
+        // Check whether the address is in the function epilog. If so, we need to
```

Comment:
Instruction analysis actually only needs to happen for the first frame. Anything beyond should be function calls which will never be inside an epilog. It might make sense to add an is_first_frame parameter to the unwinder.